### PR TITLE
initial_commit_hw_#2

### PR DESCRIPTION
**Pull request by harryji168**
this is second week homework

### CODE REVIEW

File: turtle_graphics/turtle.js [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+            }
+            return_string += row+'\n';
+        } 
+        return_string += '-- END LOG\n';
+        if(file_name ===''){
+            console.log(return_string);
+        }else{             
+            
+            fs.writeFile(file_name,return_string , function (err) {
+            if (err) return console.log(err);
```
Comment (kaylyn94) [inline turtle_graphics/turtle.js:83]: proper indentation please

File: turtle_graphics/turtle.js [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        file_name=process.argv[3].substr(9);
+    }
+
+    let run_string='';
+    if(process.argv[2].indexOf('t')){
+        run_string='flash = new Turtle()';        
+    } 
+    const arg_array = process.argv[2].split("-");  
+    for (let arg of arg_array) {   
+        if (arg.substr(0,1)  === 't') {
```
Comment (kaylyn94) [inline turtle_graphics/turtle.js:296]: alternative to `substr(0, 1)` in this case would be `startsWith` (just a suggestion of another method you could've used 😃 )

**Comment by kaylyn94** [comment]
# TURTLE GRAPHICS HOMEWORK 2
## SCORESHEET FOR STUDENT Harry

### Create a simple drawing program inspired by Turtle Graphics. 
### It will be a much simpler implementation that will only accept 
### right angles (90 degrees).

### GRADING CRITERIA:
### Course Value: 5%
### Assignment Value: 100%

### GRADE:
|Total Grade:||96|
|----------|----|----|
|Score| 96 | /100|
|Deductions| 00 |/20|


***LATE ASSIGNMENTS WILL BE DEDUCTED 5% OFF PER 24 HOURS, FOR A MAX OF 20% DEDUCTION OVERALL.
YOUR ASSIGNMENT IS CONSIDERED LATE 5 MINUTES AFTER THE DUE DATE AND TIME, UNLESS OTHERWISE DISCUSSED WITH YOUR INSTRUCTOR.***
__________________________________________________________
__________________________________________________________

### Part 1: Submitted and on Time

> Was the assignment complete and on time?

|Feature |Checkmark |Notes |
|--------|-----|------|
|Complete and submitted on time.|✅|All good!|
|On time, but incomplete.|||
|Complete, but submitted late.|||
|Incomplete and submitted late.|||
|Very late submission|||
|No submission|||

### Part 2: Features

> Was everything in assignment attempted?

|Feature |Mark |Notes |
|--------|-----|------|
|Turtle Class created.|✅||
|Constructor created with two arguments for coordinates.|✅||
|Created forward method.|✅||
|Created right method.|✅||
|Created left method.|✅||
|Created allPoints method.|✅||
|Created print method.|✅||
|Everything works as intended.|✅||

### Part 3: Stretches

> Were Stretches attempted?

|Feature |Mark |Notes |
|--------|-----|------|
|Attempted Stretch 1.|✅||
|Attempted Stretch 2.|✅||

__________________________________________________________
__________________________________________________________

### ASSIGNMENT RUBRIC

|A+|A|A-|B+|B|B-|C+|C-F|
|--------|-------|-------|-------|-------|-------|-------|------|
|90%-100%|85%-89%|80%-84%|76%-79%|72%-75%|68%-71%|64%-67%|0%-63%|
|Assignment handed in on time and complete.  All instructions were followed perfectly.  All required features were created and works correctly.  All Stretches or Challenges were completed.  Styling and design of front end meets requirements, and extra creative effort has been applied to this assignment overall.|Assignment handed in on time and complete.  All instructions were followed perfectly.  All required features were created and works correctly.  Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable. **OR** Stretches or Challenges not attempted, but creative effort has been applied to the assignment overall.|Assignment handed in on time and complete.  All instructions were followed.  All required features were created and works correctly or mostly works.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable.|Assignment handed in on time and complete.  All instructions were followed.  All required features were created or attempted.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable.|Assignment handed in on time, but handed in as incomplete.  Most instructions were followed.  Most required features were created or attempted.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements.|Assignment handed in on time, but handed in as incomplete.  Instructions were somewhat followed.  Most required features were created or attempted.  No Stretches or Challenges were attempted.  Minimal effort in styling and design of front end.|Assignment handed in on time, but handed in as incomplete.  Instructions were somewhat followed.  Some required features were created or attempted.  No Stretches or Challenges were attempted.  Minimal effort in styling and design of front end.|Minimal or no effort.Incomplete/ missing components/insufficient.|

File: turtle_graphics/turtle.js (hunk 2 above)
Comment (harryji168) [reply turtle_graphics/turtle.js:296]: It is good to know  startsWith ,the code will simple like the following, thanks.

 if(arg.startsWith('t')){    
            const pos_array = arg.substr(1).split(",");             
            run_string += " flash = new Turtle("+pos_array[0]+","+pos_array[1]+")";                        
        } else if (arg.startsWith('f')) {
            run_string +=".forward("+arg.substr(1)+")";
        } else if (arg.startsWith('r')) {
            run_string +=".right()";
        } else if (arg.startsWith('l')) {
            run_string +=".left()";
        }

File: turtle_graphics/turtle.js (hunk 1 above)
Comment (harryji168) [reply turtle_graphics/turtle.js:83]: yes, the code is copying from the internet. so forgot indentation. thanks

**Review by harryji168 — COMMENTED** [review]
thanks

**Pull request merged**